src/cbica_toolkit/src/HausdorffDistance.h

```diff
@@ -0,0 +1,92 @@
+
+// Hausdorff boundary distances, given two binary image masks
+//
+// Both fixed and moving images should have the same unsigned type
+// Only handles 3D images
+
+#ifndef _HausdorffDistance_h
+#define _HausdorffDistance_h
+
+//#include "AbstractValidationMetric.h"
+
+#include "itkImageToImageMetric.h"
+
+#include "vnl/vnl_math.h"
+
+template <class TFixedImage, class TMovingImage>
+class HausdorffDistanceImageToImageMetric :
+  public itk::ImageToImageMetric<TFixedImage, TMovingImage>/*, public AbstractValidationMetric*/
+{
+
+public:
+
+  // Validation metric properties
+  bool IsInputBinary() { return true; }
+  bool IsSymmetric() { return false; }
+  double GetBestScore() { return 0; }
+  double GetWorstScore() { return vnl_huge_val(1.0); }
+
+  /** Standard class typedefs. */
+  typedef HausdorffDistanceImageToImageMetric           Self;
+  typedef itk::ImageToImageMetric<TFixedImage, TMovingImage> Superclass;
+  typedef itk::SmartPointer<Self>                            Pointer;
+  typedef itk::SmartPointer<const Self>                      ConstPointer;
+
+  typedef TFixedImage FixedImageType;
+  typedef TMovingImage MovingImageType;
+
+  /** Method for creation through the object factory. */
+  itkNewMacro(Self);
+
+  /** Run-time type information (and related methods). */
+  itkTypeMacro(HausdorffDistanceImageToImageMetric,
+    itk::ImageToImageMetric);
+
+  typedef typename Superclass::MeasureType MeasureType;
+  typedef typename Superclass::TransformParametersType TransformParametersType;
+  typedef typename Superclass::DerivativeType DerivativeType;
+
+  typedef typename TFixedImage::Pointer FixedImagePointer;
+  typedef typename TFixedImage::IndexType FixedImageIndexType;
+  typedef typename TFixedImage::PixelType FixedImagePixelType;
+  typedef typename TFixedImage::PointType FixedImagePointType;
+  typedef typename TFixedImage::SizeType FixedImageSizeType;
+  typedef typename TFixedImage::SpacingType FixedImageSpacingType;
+
+  void SetPercentile(double p);
+
+  MeasureType GetValue() const;
+
+  MeasureType GetValue(const TransformParametersType& p) const
+  { // TODO: apply transform with nearest neighbor interpolation
+    return this->GetValue(); }
+
+  void GetDerivative(const TransformParametersType& p, DerivativeType& dp) const
+  { itkExceptionMacro(<< "Not implemented"); }
+
+  void GetValueAndDerivative(const TransformParametersType& p, MeasureType& v, DerivativeType& dp) const
+  { itkExceptionMacro(<< "Not implemented"); }
+
+  void BlurringOn() { m_DoBlurring = true; }
+  void BlurringOff() { m_DoBlurring = false; }
```

Comment:
The way to declare getters/setters in itk filter is:

```cpp
itkSetMacro(Blurring, bool); // Notice the lack of "m_"
itkGetConstMacro(Blurring, bool);

//...

bool m_Blurring = true; // If you want true as the default
```

Reply:
I am taking this class directly from ITK, so I won't make any changes to it.